core/bin/external_node/src/main.rs

```diff
@@ -363,14 +363,13 @@ async fn run_core(
     );
     app_health.insert_component(batch_status_updater.health_check())?;
 
-    let commitment_generator_pool = singleton_pool_builder
-        .build()
-        .await
-        .context("failed to build a commitment_generator_pool")?;
-    let commitment_generator = CommitmentGenerator::new(
-        commitment_generator_pool,
+    let mut commitment_generator = CommitmentGenerator::new(
+        connection_pool.clone(),
```

Comment:
Does using a common pool make sense, or would it make sense to create a separate pool for the generator? Realistically, it wouldn't require many connections unless a node is catching up.

Reply:
I think it's ok. IIRC currently in external node we use common pool for almost everything